pyccel/ast/numpyext.py

```diff
@@ -579,22 +630,39 @@ def stop(self):
         return self._stop
 
     @property
-    def size(self):
-        return self._size
+    def num(self):
+        """Represent the number of generated elements by the linspace function."""
+        return self._num
 
     @property
     def index(self):
         return self._index
 
+    @property
+    def rank(self):
+        return self._rank
+
     @property
     def step(self):
-        return (self.stop - self.start) / (self.size - 1)
+        if isinstance(self.endpoint, LiteralFalse):
+            return PyccelDiv(PyccelMinus(self.stop, self.start), self.num)
+        elif isinstance(self.endpoint, LiteralTrue):
+            PyccelDiv(PyccelMinus(self.stop, self.start), PyccelMinus(self.num, LiteralInteger(1)))
```

Comment:
i was thinking that if we know the `endpoint` is `True`, we can avoid the minus operation
I agree the check for  `LiteralTrue` does nothing since I cast the `endpoint` to `PythonInt`

Reply:
I mean line 650 does nothing. Maybe you meant to return the `PyccelDiv`?

Reply:
Oups i forgot the return

Reply:
> I agree the check for `LiteralTrue` does nothing since I cast the endpoint to `PythonInt`

? I didn't see that. Where's that done?

Reply:
in the line 651, if the endpoint was a variable and we don't know if it's True or False

Reply:
But you cast after the check so the check does do something